File: Data Structures & Algorithms/house-robber/submission-0.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        vector<int> ans(nums.size());
        for (int i = 0; i < nums.size(); i++) {
            if (i == 0) {
                ans[i] = nums[i];
            } else if (i == 1) {
                ans[i] = (nums[i] > nums[i-1]) ? (nums[i]) : (nums[i-1]);
            } else {
                ans[i] = ((nums[i] + ans[i -2]) > (ans[i - 1]) ? (nums[i] + ans[i -2]) : (ans[i - 1]));
            }
        }
        return ans[nums.size() - 1];
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        #to store cout of each num
        counts = {}
        buckets = [[] for i in range(0, len(nums) + 1)]
        for number in nums:
            if number in counts:
                counts[number] += 1
            else :
                counts[number] = 1
        #should have a hashmap with each number's frequency
        for key, value in counts.items():
            buckets[value].append(key)
        
        i = len(buckets) - 1
        output = []
        while (i >= 0):
            for number in buckets[i]:
                output.append(number)
            if len(output) == k:
                return output
            i-=1;